backend/src/main/java/ai/verta/modeldb/versioning/VersioningServiceImpl.java

```diff
@@ -242,11 +253,23 @@ public void deleteRepository(
   public void listCommits(
       ListCommitsRequest request, StreamObserver<ListCommitsRequest.Response> responseObserver) {
     try {
+      Repository repository =
+          repositoryDAO
+              .getRepository(
+                  GetRepositoryRequest.newBuilder().setId(request.getRepositoryId()).build())
+              .getRepository();
```

Comment:
Instead of fetching whole repository we should have to fetch entity resource from UAC and get workspace_id from it.

Reply:
while getting repository using `getRepository` it will also call UAC for entity resource. so instead calling DB and UAC both just get entity resource from UAC is better.